src/gui/owncloudgui.cpp

```diff
@@ -56,16 +56,40 @@ namespace OCC {
 
 const char propertyAccountC[] = "oc_account";
 
+ownCloudGui *ownCloudGui::_instance = nullptr;
+
+ownCloudGui *ownCloudGui::instance(Application *parent)
+{
+    if (!_instance) {
+        Q_ASSERT(parent);
+        _instance = new ownCloudGui(parent);
+    }
+    return _instance;
+}
+
 ownCloudGui::ownCloudGui(Application *parent)
     : QObject(parent)
     , _tray(nullptr)
-    , _settingsDialog(new SettingsDialog(this))
+    , _settingsDialog(nullptr)
```

Comment:
Good idea, better get this initialization delayed indeed!

Reply:
Agreed, see next comment ;-)